client/components/post-schedule/utils.js

```diff
@@ -1,62 +1,108 @@
+import { moment } from 'i18n-calypso';
 
-export default {
-	isValidGMTOffset( gmtOffset ) {
-		return 'number' === typeof gmtOffset;
-	},
-
-	/**
-	 * Return localized date depending of given timezone and gmtOffset
-	 * parameters.
-	 *
-	 * @param {Moment} date - date instance
-	 * @param {String} tz - timezone
-	 * @param {Number} gmt - gmt offset
-	 * @return {Moment} localized date
-	 */
-	getLocalizedDate( date, tz, gmt ) {
-		date = i18n.moment( date );
-
-		if ( tz ) {
-			date.tz( tz );
-		} else if ( this.isValidGMTOffset( gmt ) ) {
-			date.utcOffset( gmt );
-		}
-
-		return date;
-	},
-
-	convertDateToUserLocation( date, tz, gmt ) {
-		if ( ! ( tz || this.isValidGMTOffset( gmt ) ) ) {
-			return i18n.moment( date );
-		}
-
-		return this.getDateInLocalUTC( date )
-			.subtract( this.getTimeOffset( date, tz, gmt ), 'minute' );
-	},
-
-	convertDateToGivenOffset( date, tz, gmt ) {
-		date = this.getLocalizedDate( date, tz, gmt )
-			.add( this.getTimeOffset( date, tz, gmt ), 'minute' );
-
-		if ( ! tz && this.isValidGMTOffset( gmt ) ) {
-			date.utcOffset( gmt );
-		};
-
-		return date;
-	},
-
-	getTimeOffset( date, tz, gmt ) {
-		const userLocalDate = this.getDateInLocalUTC( date ),
-			localizedDate = this.getLocalizedDate( date, tz, gmt );
-
-		return userLocalDate.utcOffset() - localizedDate.utcOffset();
-	},
-
-	getDateInLocalUTC( date ) {
-		return i18n.moment( date.format ? date.format() : date )
+/**
+ * Check whether is a valid gmtOffset value.
+ * Basically it should be a number.
+ *
+ * @param  {*}  gmtOffset - gmt offset
+ * @return {Boolean} is it a valid gtm offset?
+ */
+const isValidGMTOffset = gmtOffset => 'number' === typeof gmtOffset;
+
+/**
+ * Return localized date depending of given timezone or gmtOffset
+ * parameters.
+ *
+ * @param {Moment} date - date instance
+ * @param {String} tz - timezone
+ * @param {Number} gmt - gmt offset in minutes
+ * @return {Moment} localized date
+ */
+const getLocalizedDate = ( date, tz, gmt ) => {
+	date = moment( date );
+
+	if ( tz ) {
+		date.tz( tz );
+	} else if ( isValidGMTOffset( gmt ) ) {
+		date.utcOffset( gmt );
+	}
+
+	return date;
+};
+
+const getDateInLocalUTC = date => moment( date.format ? date.format() : date );
+
+const getTimeOffset = ( date, tz, gmt ) => {
+	const userLocalDate = getDateInLocalUTC( date );
+	const localizedDate = getLocalizedDate( date, tz, gmt );
+
+	return userLocalDate.utcOffset() - localizedDate.utcOffset();
+};
+
+const convertDateToUserLocation = ( date, tz, gmt ) => {
+	if ( ! ( tz || isValidGMTOffset( gmt ) ) ) {
+		return moment( date );
 	}
-}
+
+	return getDateInLocalUTC( date )
+		.subtract( getTimeOffset( date, tz, gmt ), 'minute' );
+};
+
+const convertDateToGivenOffset = ( date, tz, gmt ) => {
+	date = getLocalizedDate( date, tz, gmt )
+		.add( getTimeOffset( date, tz, gmt ), 'minute' );
+
+	if ( ! tz && isValidGMTOffset( gmt ) ) {
+		date.utcOffset( gmt );
+	}
+
+	return date;
+};
+
+const convertMinutesToHHMM = minutes => {
```

Comment:
The helper function expects some minutes instead of a Date (Moment) object. But we could create a temporal instance just to represent the part of `hh:mm`. We should tweak it to handle the negative signal, though.

```es6
const convertMinutesToHHMM = ( minutes = 0 ) => {
	const tempoDate = moment( '2000-1-1').minutes( Math.abs( minutes ) );
	return ( minutes > 0 ? '' : '-' ) + tempoDate.format( 'hh:mm' );
}
```